src/Util/JsonLD.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/**
+ * @file src/Util/JsonLD.php
+ */
+namespace Friendica\Util;
+
+use Friendica\Core\Cache;
+use digitalbazaar\jsonld as DBJsonLD;
+use Exception;
+
+/**
+ * @brief This class contain methods to work with JsonLD data
+ */
+class JsonLD
+{
+	/**
+	 * @brief Loader for LD-JSON validation
+	 *
+	 * @param $url
+	 *
+	 * @return the loaded data
+	 */
+	public static function documentLoader($url)
+	{
+		$recursion = 0;
+
+		$x = debug_backtrace();
+		if ($x) {
+			foreach ($x as $n) {
+				if ($n['function'] === __FUNCTION__)  {
+					$recursion ++;
+				}
+			}
+		}
+
+		if ($recursion > 5) {
+			logger('jsonld bomb detected at: ' . $url);
+			exit();
+		}
+
+		$result = Cache::get('documentLoader:' . $url);
+		if (!is_null($result)) {
+			return $result;
+		}
+
+		$data = jsonld_default_document_loader($url);
+		Cache::set('documentLoader:' . $url, $data, CACHE_DAY);
```

Comment:
Is this used to query remote users' outboxes? If yes, isn't the cache duration too long?

Reply:
It is used for storing JSON-LD definitions. This is something that mostly never changes and we even could increase the cache duration to a month or so.